mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,382 @@
+        )
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts,
+            temp_dir,
+            plot_feature_importance,
+            run_id,
+            "shap_feature_importance",
+            dataset_name,
+            model,
+        )
+
+    def _evaluate_classifier(self, temp_dir, model, X, y, dataset_name, feature_names, run_id, evaluator_config):
+        from mlflow.models.evaluation.lift_curve import plot_lift_curve
+
+        # Note: require labels to be number of 0, 1, 2, .. num_classes - 1
+        label_list = sorted(list(set(y)))
+        assert label_list[0] == 0, "Label values must being at '0'."
+        num_classes = len(label_list)
+
+        y_pred = model.predict(X)
+
+        is_binomial = num_classes <= 2
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+        metrics["example_count"] = len(X)
+
+        # TODO: sum/mean on what data ?
+        #  [P0] Sum: Computes the (weighted) sum of the given values.
+        #  [P0] Mean: Computes the (weighted) mean of the given values.
+
+        if is_binomial:
+            if hasattr(model, "predict_proba"):
+                y_probs = model.predict_proba(X)
+                y_prob = y_probs[:, 1]
+            else:
+                y_probs = None
+                y_prob = None
+
+            confusion_matrix = sk_metrics.confusion_matrix(y, y_pred)
+            tn, fp, fn, tp = confusion_matrix.ravel()
+            metrics["true_negatives"] = tn
+            metrics["false_positives"] = fp
+            metrics["false_negatives"] = fn
+            metrics["true_positives"] = tp
+            metrics["recall"] = sk_metrics.recall_score(y, y_pred)
+            metrics["precision"] = sk_metrics.precision_score(y, y_pred)
+            metrics["f1_score"] = sk_metrics.f1_score(y, y_pred)
+
+            # TODO:
+            #  compute hinge loss, this requires calling decision_function of the model
+            #  e.g., see https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html#sklearn.svm.LinearSVC.decision_function
+            if y_probs is not None:
+                fpr, tpr, thresholds = sk_metrics.roc_curve(y, y_prob)
+                roc_curve_pandas_df = pd.DataFrame(
+                    {"fpr": fpr, "tpr": tpr, "thresholds": thresholds}
+                )
+                self._log_pandas_df_artifact(
+                    artifacts, temp_dir, roc_curve_pandas_df, run_id, "roc_curve", dataset_name, model,
+                )
+
+                roc_auc = sk_metrics.auc(fpr, tpr)
+                metrics["roc_auc"] = roc_auc
+
+                def plot_roc_curve():
+                    sk_metrics.RocCurveDisplay(
+                        fpr=fpr, tpr=tpr, roc_auc=roc_auc,
+                    ).plot()
+
+                if hasattr(sk_metrics, 'RocCurveDisplay'):
```

Comment:
The sklearn >= 0.24.1 support XXXDisplay, and note that MLR 10.2 already preinstall sklearn 0.24.1